api/python/provisioner/commands/destroy_vm.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from typing import Type
+import logging
+
+from .. import (
+    inputs,
+    errors,
+    config,
+    ALL_MINIONS
+)
+from ..utils import run_subprocess_cmd
+from ..salt import local_minion_id
+from ..pillar import (
+    PillarKey,
+    PillarResolver
+)
+from ..vendor import attr
+# TODO IMPROVE EOS-8473
+
+from .configure_setup import (
+    SetupType
+)
+from .deploy import (
+    build_deploy_run_args,
+    Deploy
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+deploy_states = dict(
+    prvsnr=[
+        "system.storage.glusterfs.teardown.volume_remove",
+        "system.storage.glusterfs.teardown.remove_dir",
+        "system.storage.glusterfs.teardown.services",
+        "system.storage.glusterfs.teardown.package_remove"
+    ]
+)
+
+run_args_type = build_deploy_run_args(deploy_states)
+
+
+@attr.s(auto_attribs=True)
+class DestroyVM(Deploy):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = run_args_type
+
+    def _remove_prvsnr(self, nodes):
+        for node in nodes:
+            if local_minion_id() == node:
+                logger.info(f"remove cortx-prvsnr rpms on {node}")
+                run_subprocess_cmd(
+                    f"ssh {node} yum erase -y cortx-prvsnr")
+            else:
+                logger.info(
+                    f"remove cortx-prvsnr python36-cortx-prvsnr on {node}")
+                run_subprocess_cmd(
+                    f"ssh {node} yum erase -y"
+                    "cortx-prvsnr python36-cortx-prvsnr")
+
+    def _remove_passwordless_ssh(self, nodes):
+        for node in nodes:
+            logger.info(f"Remove passwordless configuration with {node}")
+            run_subprocess_cmd(
+                f"rm -rf /root/.ssh")
+
+    def _remove_salt(self, nodes):
+        salt_cache_dir = '/var/cache/salt'
```

Comment:
in case of glusterfs `/var/cache/salt/master/jobs` would be a mount point. so we should umount that first. Logic should likely be:
- stop/uninstall salt
- umount glusterfs
- cleanup salt dirs
- stop and cleanup gluster

Reply:
If we remove salt first then we cannot use salt states to teardown glusterfs. We need to use manual steps/commands to unmount, remove volume , remove bricks dir etc.

I followed steps given in Wiki:
`https://github.com/Seagate/cortx-prvsnr/wiki/Auto-Deploy-VM-Provisioner-CLI-Commands#teardown-provisioner-bootstrapped-environment`